src/azul/drs.py

```diff
@@ -269,21 +268,38 @@ def get_object(self,
         depends on the access method specified.
         """
         if access_id is None:
-            return self._get_object(object_id, access_method=access_method)
+            return self._get_object(drs_uri, access_method)
         else:
-            return self._get_object_access(object_id, access_id, access_method=access_method)
+            return self._get_object_access(drs_uri, access_id, access_method)
 
-    def _get_object(self, object_id: str, access_method: AccessMethod) -> str:
-        url = self.url / object_id
+    def _get_drs_url(self, drs_uri: str, access_id: Optional[str] = None) -> str:
+        """
+        Translate a DRS URI into a DRS URL. All query params included in the DRS
+        URI (eg '{drs_uri}?version=123') will be carried over to the DRS URL.
+        Only hostname-based DRS URIs (drs://<hostname>/<id>) are supported while
+        compact, identifier-based URIs (drs://[provider_code/]namespace:accession)
+        are not.
+        """
+        parsed = furl(drs_uri)
+        require(parsed.scheme == 'drs',
+                f'The DRS URI {drs_uri} does not have the correct scheme')
+        reject(':' in parsed.netloc or ':' in str(parsed.path),
+               f'The value {drs_uri} is not a valid Hostname-based DRS URI')
```

Comment:
```suggestion
               f'The value {drs_uri} is not a valid hostname-based DRS URI')
```

When I requested the corresponding change to the docstring, you should have picked this up on your own.